File: leetcode/0202-happy-number/solution.py
class Solution(object):
    def isHappy(self, n):
        n=str(n)
        j=0
        while j<100:
            d={}
            for i in range(0,len(n)):
                d[i]=int(n[i])**2
            s=sum(d.values())
            x=False
            if s==1:
                x=True
                break
            else:
                n=str(s)
            j+=1
        return x
        """
        :type n: int
        :rtype: bool
        """
        

File: leetcode/0202-happy-number/solution.cpp
class Solution {
public:
    bool isHappy(int n) {
        int num=n;
        for(int i=0;i<10;i++){
            vector<int> digits;
            while(num!=0){
                digits.push_back(num%10);
                num=num/10;
            }
            for(int i:digits){
                num=num+ i*i;
            }
            n=num;
        }
        if (n==1){
            return true;
        }
        return false;
    }
};
